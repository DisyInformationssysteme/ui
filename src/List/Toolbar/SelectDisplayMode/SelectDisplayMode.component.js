import React, { PropTypes } from 'react';
import { Nav, NavDropdown, MenuItem } from 'react-bootstrap';
import Icon from '../../../Icon';

function getIcon(selected) {
	switch (selected) {
	case 'table': return 'fa-bars';
	case 'large': return 'fa-list-alt';
	case 'tile': return 'fa-th-large';
	default: return 'fa-bars';
	}
}

function getLabel(selected) {
	switch (selected) {
	case 'table': return 'Table';
	case 'large': return 'Expanded';
	case 'tile': return 'Tile';
	default: return 'Table';
	}
}

const displayModes = ['table', 'large', 'tile'];

/**
 * @param {object} props react props
 */
function SelectDisplayMode(props) {
	const selected = props.selected ? props.selected : 'table';
	const displayIcon = (<Icon name={getIcon(selected)} />);
	const onSelectDisplayMode = (value, e) => {
		props.onSelectDisplayMode(e, value);
	};
	return (
		<Nav>
			<label className="navbar-text">Display:</label>
			<NavDropdown title={displayIcon} id="grid" onSelect={onSelectDisplayMode}>
				{displayModes.map(mode => (
					<MenuItem eventKey={mode}><Icon name={getIcon(mode)} /> {getLabel(mode)}</MenuItem>

				))}
			</NavDropdown>
		</Nav>
	);
}
SelectDisplayMode.propTypes = {
	onSelectDisplayMode: PropTypes.func,
	selected: PropTypes.string,
};

export default SelectDisplayMode;
