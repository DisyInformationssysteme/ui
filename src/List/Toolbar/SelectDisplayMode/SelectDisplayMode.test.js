import React from 'react';
import renderer from 'react-test-renderer';

import SelectDisplayMode from './SelectDisplayMode.component';

jest.mock('react-dom');

describe('SelectDisplayMode', () => {
	it('should render', () => {
		const props = {
			onSelectDisplayMode: jest.fn(),
		};
		const wrapper = renderer.create(
			<SelectDisplayMode {...props} />
		).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
	it('should render with selected = table', () => {
		const props = {
			onSelectDisplayMode: jest.fn(),
			selected: 'table',
		};
		const wrapper = renderer.create(
			<SelectDisplayMode {...props} />
		).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
	it('should render with selected = large', () => {
		const props = {
			onSelectDisplayMode: jest.fn(),
			selected: 'large',
		};
		const wrapper = renderer.create(
			<SelectDisplayMode {...props} />
		).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
	it('should render with selected = tile', () => {
		const props = {
			onSelectDisplayMode: jest.fn(),
			selected: 'tile',
		};
		const wrapper = renderer.create(
			<SelectDisplayMode {...props} />
		).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
});
