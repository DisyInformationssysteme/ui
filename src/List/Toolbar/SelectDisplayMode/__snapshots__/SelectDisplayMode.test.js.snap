exports[`SelectDisplayMode should render 1`] = `
<ul
  className="nav"
  role={null}>
  <label
    active={undefined}
    activeHref={undefined}
    activeKey={undefined}
    className="navbar-text"
    onSelect={null}>
    Display:
  </label>
  <li
    className="dropdown"
    style={undefined}>
    <a
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle"
      href=""
      id="grid"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button">
      <i
        aria-hidden="true"
        className="fa fa-bars"
        title={null} />
       
      <span
        className="caret" />
    </a>
    <ul
      aria-labelledby="grid"
      className="dropdown-menu"
      role="menu">
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          <i
            aria-hidden="true"
            className="fa fa-bars"
            title={null} />
           
          Table
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          <i
            aria-hidden="true"
            className="fa fa-list-alt"
            title={null} />
           
          Expanded
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          <i
            aria-hidden="true"
            className="fa fa-th-large"
            title={null} />
           
          Tile
        </a>
      </li>
    </ul>
  </li>
</ul>
`;

exports[`SelectDisplayMode should render with selected = large 1`] = `
<ul
  className="nav"
  role={null}>
  <label
    active={undefined}
    activeHref={undefined}
    activeKey={undefined}
    className="navbar-text"
    onSelect={null}>
    Display:
  </label>
  <li
    className="dropdown"
    style={undefined}>
    <a
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle"
      href=""
      id="grid"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button">
      <i
        aria-hidden="true"
        className="fa fa-list-alt"
        title={null} />
       
      <span
        className="caret" />
    </a>
    <ul
      aria-labelledby="grid"
      className="dropdown-menu"
      role="menu">
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          <i
            aria-hidden="true"
            className="fa fa-bars"
            title={null} />
           
          Table
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          <i
            aria-hidden="true"
            className="fa fa-list-alt"
            title={null} />
           
          Expanded
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          <i
            aria-hidden="true"
            className="fa fa-th-large"
            title={null} />
           
          Tile
        </a>
      </li>
    </ul>
  </li>
</ul>
`;

exports[`SelectDisplayMode should render with selected = table 1`] = `
<ul
  className="nav"
  role={null}>
  <label
    active={undefined}
    activeHref={undefined}
    activeKey={undefined}
    className="navbar-text"
    onSelect={null}>
    Display:
  </label>
  <li
    className="dropdown"
    style={undefined}>
    <a
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle"
      href=""
      id="grid"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button">
      <i
        aria-hidden="true"
        className="fa fa-bars"
        title={null} />
       
      <span
        className="caret" />
    </a>
    <ul
      aria-labelledby="grid"
      className="dropdown-menu"
      role="menu">
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          <i
            aria-hidden="true"
            className="fa fa-bars"
            title={null} />
           
          Table
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          <i
            aria-hidden="true"
            className="fa fa-list-alt"
            title={null} />
           
          Expanded
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          <i
            aria-hidden="true"
            className="fa fa-th-large"
            title={null} />
           
          Tile
        </a>
      </li>
    </ul>
  </li>
</ul>
`;

exports[`SelectDisplayMode should render with selected = tile 1`] = `
<ul
  className="nav"
  role={null}>
  <label
    active={undefined}
    activeHref={undefined}
    activeKey={undefined}
    className="navbar-text"
    onSelect={null}>
    Display:
  </label>
  <li
    className="dropdown"
    style={undefined}>
    <a
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle"
      href=""
      id="grid"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button">
      <i
        aria-hidden="true"
        className="fa fa-th-large"
        title={null} />
       
      <span
        className="caret" />
    </a>
    <ul
      aria-labelledby="grid"
      className="dropdown-menu"
      role="menu">
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          <i
            aria-hidden="true"
            className="fa fa-bars"
            title={null} />
           
          Table
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          <i
            aria-hidden="true"
            className="fa fa-list-alt"
            title={null} />
           
          Expanded
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          <i
            aria-hidden="true"
            className="fa fa-th-large"
            title={null} />
           
          Tile
        </a>
      </li>
    </ul>
  </li>
</ul>
`;
